test/run/test_stages.py

```diff
@@ -139,6 +149,30 @@ def raise_assertion(msg):
             )
             raise AssertionError(f"{msg}\n\n{diff}")
 
+        def remove_items(lst, condition):
+            """
+            Removes items from a list in-place based on a condition
+            :return: True if at least one item was removed, false otherwise
+            """
+            original_len = len(lst)
+            lst[:] = [item for item in lst if not condition(item)]
+            return original_len != len(lst)
+
+        def path_equal_or_is_descendant(path, potential_ancestor):
+            """
+            :return: Returns true if path == potential_ancestor or if path is inside potential_ancestor
+            """
+            return path == potential_ancestor or path.startswith(potential_ancestor + "/")
+
+        for added_dir in tree_diff1.get('added_directories', []):
+            was_deleted = remove_items(
```

Comment:
Perhaps too much of a stickler for this but I'd much prefer something like this to the in-place method.

```python
original = tree_diff2['added_files']
filtered = [p for p in original if not ...]

if len(original) != len(filtered):
    raise_assertion(...)

tree_diff2['added_files'] = filtered
```

Reply:
Much cleaner, thanks! :)